main.py

```diff
@@ -740,13 +740,21 @@ def ui_access_wrapper(self, *args, **kwargs):
 
     get_redirect_route(
         r'%s' % feconf.CSRF_HANDLER_URL, base.CsrfTokenHandler),
+]
 
-    get_redirect_route(r'/<classroom_name>', classroom.ClassroomPage),
-    get_redirect_route(
-        r'/<subject>/<topic>', custom_landing_pages.TopicLandingPage),
+# Adding redirects for classroom pages.
+for classroom_name in feconf.CLASSROOM_PAGES:
+    URLS.append(
+        get_redirect_route(r'/%s' % classroom_name, classroom.ClassroomPage))
 
-    get_redirect_route(r'/<:.*>', base.Error404Handler),
-]
+# Adding redirects for topic landing pages.
+for subject in feconf.AVAILABLE_LANDING_PAGES:
+    URLS.append(
+        get_redirect_route(
+            r'/%s/<topic>' % subject, custom_landing_pages.TopicLandingPage))
```

Comment:
It seems to me like you don't need to interpolate "topic" here either -- you can get that from the constant and enumerate explicitly too?

Reply:
Done.